geonode_mapstore_client/client/js/epics/gnsave.js

```diff
@@ -169,13 +179,16 @@ export const gnSaveContent = (action$, store) =>
                         window.location.reload();
                         return Observable.empty();
                     }
+                    const dimensions = timeseries?.has_time ? getDimensions({...currentResource, has_time: true}) : [];
                     return Observable.of(
                         saveSuccess(resource),
                         setResource({
                             ...currentResource,
                             ...body,
-                            ...resource
+                            ...resource,
+                            timeseries
                         }),
+                        ...(timeseries ? [updateNode(layerId, 'layers', { dimensions: dimensions?.length > 0 ? dimensions : undefined })] : []),
```

Comment:
Please move all the code inside the DATASET SaveAPI above, the gnSaveContent epic should be generic

Reply:
Updated